### PR TITLE
Add support to RecordBatch protocol

### PR DESCRIPTION
**Pull request by tulios**
I've tested with a custom implementation of the producer API, and `kafkacat` managed to consume the messages

### CODE REVIEW

**Comment by tulios** [comment]
The producer v3 tests are failing on Linux, similar to the GZIP issues I had with Mac OS; I'll have to generate specific fixtures for it

screen shot 2018-05-31 at 07 09 25
screen shot 2018-05-31 at 07 09 16

File: src/protocol/recordBatch/crc32C/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  test('perform CRC32C computations', () => {
+    const longString =
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi mollis cursus metus vel tristique. Proin congue massa massa, a malesuada dolor ullamcorper a. Nulla eget leo vel orci venenatis placerat. Donec semper condimentum justo, vel sollicitudin dolor consequat id. Nunc sed aliquet felis, eget congue nisi. Mauris eu justo suscipit, elementum turpis ut, molestie tellus. Mauris ornare rutrum fringilla. Nulla dignissim luctus pretium. Nullam nec eros hendrerit sapien pellentesque sollicitudin. Integer eget ligula dui. Mauris nec cursus nibh. Nunc interdum elementum leo, eu sagittis eros sodales nec. Duis dictum nulla sed tincidunt malesuada. Quisque in vulputate sapien. Sed sit amet tellus a est porta rhoncus sed eu metus. Mauris non pulvinar nisl, volutpat luctus enim. Suspendisse est nisi, sagittis at risus quis, ultricies rhoncus sem. Donec ullamcorper purus eget sapien facilisis, eu eleifend felis viverra. Suspendisse elit neque, semper aliquet neque sed, egestas tempus leo. Duis condimentum turpis duis.'
+    const buffer = Buffer.from(longString)
+    expect(crc32C(buffer)).toEqual(1796588439)
```
Comment (Nevon) [inline src/protocol/recordBatch/crc32C/index.spec.js]: How did you get this number?

**Comment by Nevon** [comment]
How come you get different output on different platforms? That seems like a not great thing.

**Comment by tulios** [comment]
I think it was just missing the response tests; it's green now. I haven't done anything special. Commit 6e74c37 fixed the problem.

**Pull request merged**